CMakeLists.txt

```diff
@@ -33,37 +47,77 @@ CPMAddPackage(
   OPTIONS "FMT_INSTALL YES" # create an installable target
 )
 
+# ---- Setup default project_options ----
+
+# include(DynamicProjectOptions)
+# # Initialize project_options variable related to this project
+# # This overwrites `project_options` and sets `project_warnings`
+# # uncomment the options to enable them:
+# dynamic_project_options(
+#   # MSVC_WARNINGS "/permissive-" # Override the defaults for the MSVC warnings
+#   # CLANG_WARNINGS "-Wall;-Wextra" # Override the defaults for the CLANG warnings
+#   # GCC_WARNINGS "-Wextra" # Override the defaults for the GCC warnings
+# )
+# target_compile_features(project_options INTERFACE cxx_std_${CMAKE_CXX_STANDARD})
+
 # ---- Add source files ----
 
 # Note: globbing sources is considered bad practice as CMake's generators may not detect new files
 # automatically. Keep that in mind when changing files, or explicitly mention them here.
-file(GLOB_RECURSE headers CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/include/*.h")
-file(GLOB_RECURSE sources CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/source/*.cpp")
+# XXX file(GLOB_RECURSE headers CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/include/*.h")
+# XXX file(GLOB_RECURSE sources CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/source/*.cpp")
+set(headers source/greeter.cpp)
+set(sources include/greeter/greeter.h)
 
 # ---- Create library ----
 
 # Note: for header-only libraries change all PUBLIC flags to INTERFACE and create an interface
 # target: add_library(${PROJECT_NAME} INTERFACE)
 add_library(${PROJECT_NAME} ${headers} ${sources})
+add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
 target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_20)
 
-# enable compiler warnings
-include(cmake/WarningsAsErrors.cmake)
-
 # being a cross-platform target, we enforce standards conformance on MSVC
 target_compile_options(${PROJECT_NAME} PRIVATE "$<$<COMPILE_LANG_AND_ID:CXX,MSVC>:/permissive->")
 
 # Link dependencies
 target_link_libraries(${PROJECT_NAME} PUBLIC fmt::fmt)
 
+# FIXME: should be PRIVATE! CK
+# TODO: target_link_libraries(${PROJECT_NAME} PUBLIC project_warnings)
+# TODO: target_link_libraries(${PROJECT_NAME} PUBLIC project_options)
+
+target_include_directories(${PROJECT_NAME} PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>)
+# target_include_directories with the SYSTEM modifier will request the compiler
+# to omit warnings from the provided paths, if the compiler supports that
 target_include_directories(
   ${PROJECT_NAME} SYSTEM PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
-                                $<INSTALL_INTERFACE:include/${PROJECT_NAME}-${PROJECT_VERSION}>
+                                $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
 )
 
+if(CMAKE_SKIP_INSTALL_RULES)
+  return()
+endif()
+
+if(NOT PROJECT_IS_TOP_LEVEL)
```

Comment:
**TODO**: should it possible to install developer versions?

It is not a good idea to export `code_coverage` or `sanitizer` compiler options!